src/cgns_f.F90

```diff
@@ -4580,26 +4581,70 @@ END SUBROUTINE cgio_set_label_f
 !*---------------------------------------------------------
 !!$!!$     SUBROUTINE cgio_write_all_data_f(
 !!$    cgint_f *cgio_num, double *id, void *data, cgint_f *ier) BIND(C,NAME="")
-!!$  
+!!$
 !*---------------------------------------------------------
 !!$!!$     SUBROUTINE cgio_write_block_data_f(
 !!$    cgint_f *cgio_num, double *id, cgsize_t *b_start, cgsize_t *b_end,
 !!$    void *data, cgint_f *ier) BIND(C,NAME="")
 !!$
-!!$  
+!!$
 !*---------------------------------------------------------
 !!$!!$     SUBROUTINE cgio_write_data_f(
 !!$    cgint_f *cgio_num, double *id, cgsize_t *s_start, cgsize_t *s_end,
 !!$    cgsize_t *s_stride, cgsize_t *m_ndims, cgsize_t *m_dims, cgsize_t *m_start,
 !!$    cgsize_t *m_end, cgsize_t *m_stride, void *data, cgint_f *ier) BIND(C,NAME="")
 !!$
-!!$  
+!!$
   END INTERFACE
 
   PRIVATE cg_configure_ptr
 
 CONTAINS
 
+#if CG_BUILD_PARALLEL_F
+
+  SUBROUTINE cgp_poly_section_write_f(fn, B , Z, sectionname, type, start, end, maxoffset, nbndry, S, ier)
+    IMPORT :: cgsize_t, cgenum_t
+    IMPLICIT NONE
+    INTEGER, INTENT(IN) :: fn
+    INTEGER, INTENT(IN) :: B
+    INTEGER, INTENT(IN) :: Z
+    CHARACTER(LEN=*), INTENT(IN) :: sectionname
+    INTEGER(cgenum_t), INTENT(IN) :: type
+    INTEGER(cgsize_t), INTENT(IN) :: start
+    INTEGER(cgsize_t), INTENT(IN) :: end
+    INTEGER(cgsize_t), INTENT(IN) :: maxoffset
+    INTEGER, INTENT(IN) :: nbndry
+    INTEGER, INTENT(OUT) :: S
+    INTEGER, INTENT(OUT) :: ier
+
+    CHARACTER(LEN=LEN_TRIM(sectionname)+1,KIND=C_CHAR) :: c_name
+
+    INTERFACE
+      INTEGER FUNCTION cgp_poly_section_write(fn, B , Z, sectionname, type, start, end, maxoffset, nbndry, S) &
+      BIND(C, name="cgp_poly_section_write")
+      IMPORT :: c_int, c_char, cgsize_t, cgenum_t
+      IMPLICIT NONE
+      INTEGER(c_int) :: fn
+      INTEGER(c_int) :: B
+      INTEGER(c_int) :: Z
+      CHARACTER(KIND=C_CHAR), DIMENSION(*) :: sectionname
+      INTEGER(cgenum_t) :: type
+      INTEGER(cgsize_t) :: start
+      INTEGER(cgsize_t) :: end
+      INTEGER(cgsize_t) :: maxoffset
+      INTEGER(c_int) :: nbndry
+      INTEGER(c_int) :: S
+      END FUNCTION cgp_poly_section_write
+    END INTERFACE
+
+    c_name = TRIM(sectionname)//C_NULL_CHAR
+    ier = cgp_poly_section_write(fn, B, Z, c_name, type, start, end, maxoffset, nbndry, S)
```

Comment:
You will have to, for example, INT(fn, C_INT).

Reply:
Okay